File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
def solution(s, skip, index):
    answer = ''
    skip_ch = list(skip)
    for ch in s:
        cnt = idx = 1
        start = ch
        while cnt <= index:
            start = chr(ord(start) + 1) if start < 'z' else 'a'
            if start in skip_ch:
                continue
            cnt += 1
        answer += start
    return answer

File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.java
import java.util.*;

class Solution {
    public String solution(String s, String skip, int index) {
        String answer = "";
        Map<Character, Boolean> map = new HashMap<>();
        for(int i=0 ; i<skip.length() ; i++){
            map.put(skip.charAt(i), true);
        }
        
        for(int i=0 ; i<s.length(); i++){
            int cnt = 0;
            char ch = s.charAt(i);
            while(cnt < index){
                ch = (char)(1 + (int)ch);
                if(ch > 'z')ch = (char)(ch - 'z' + 'a' - 1);
                if(map.containsKey(ch))continue;
                cnt++;
            }
            answer += ch;
        }
        return answer;
    }
}